internal/controllers/image_controller.go

```diff
@@ -223,11 +223,44 @@ func (r *ImageReconciler) deleteImage(ctx context.Context, log logr.Logger, ioCt
 		return nil
 	}
 
+	img, err := librbd.OpenImage(ioCtx, ImageIDToRBDID(image.ID), librbd.NoSnapshot)
+	if err != nil {
+		if !errors.Is(err, librbd.ErrNotFound) {
+			return fmt.Errorf("failed to open image: %w", err)
+		}
+		return nil
+	}
+
+	pools, imgs, err := img.ListChildren()
+	if err != nil {
+		return fmt.Errorf("unable to list volume image children: %w", err)
+	}
+	log.V(2).Info("Volume image references", "pools", len(pools), "rbd-images", len(imgs))
+
+	if err := img.Close(); err != nil {
+		return fmt.Errorf("unable to close image: %w", err)
+	}
```

Comment:
Potential memory leak: We call `librbd.OpenImage` and don't close it in case we error out with `img.ListChildren`.
Can you pls. put this in an extra method and use defer to close the image?